test/unit/infrastructure/adapters/GitCasWarpStateCacheAdapter.test.ts

```
@@ -195,27 +261,59 @@ describe('GitCasWarpStateCacheAdapter', () => {
       expect(idx2).toEqual({ schemaVersion: 1, snapshots: {} });
     });
 
-    it('returns empty index on invalid blob JSON or invalid schemaVersion', async () => {
+    it('fails closed on unreadable or invalid persisted indexes', async () => {
       persistence.readRef.mockResolvedValue('blob-1');
       persistence.readBlob.mockResolvedValueOnce(new TextEncoder().encode('invalid-json'));
-      expect(await adapter._readIndex()).toEqual({ schemaVersion: 1, snapshots: {} });
+      await expect(adapter._readIndex()).rejects.toThrow(/malformed state-cache index/);
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)